deployment/ccip/deploy_home_chain.go

```diff
@@ -85,7 +86,24 @@ func MustABIEncode(abiString string, args ...interface{}) []byte {
 	return encoded
 }
 
-func DeployCapReg(lggr logger.Logger, ab deployment.AddressBook, chain deployment.Chain) (*ContractDeploy[*capabilities_registry.CapabilitiesRegistry], error) {
+// DeployCapReg deploys the CapabilitiesRegistry contract if it is not already deployed
+// and returns a ContractDeploy struct with the address and contract instance.
+func DeployCapReg(
+	lggr logger.Logger,
+	ab deployment.AddressBook,
+	chain deployment.Chain,
+) (*ContractDeploy[*capabilities_registry.CapabilitiesRegistry], error) {
+	capRegAddress, err := deployment.SearchAddressBook(ab, chain.Selector, CapabilitiesRegistry)
```

Comment:
Hmm, not sure about that. So if `error` is nil we assume we found it, but if `error` is present we assume it's not there and needs deployment? What if the error is not related to a lack of capabilities registry but some other transient issue that would trigger unnecessary deployment? 

I would consider a more explicit interface here, for instance, something like this
```go
// bool added to returned types
func SearchAddressBook(ab AddressBook, chain uint64, typ ContractType) (string, bool, error) {}

capRegAddress, ok, err := deployment.SearchAddressBook(ab, chain.Selector, CapabilitiesRegistry)
if err != nil {
    return nil, err
}
if !ok {
   deployContract
} else {
   returnAlreadyDeployed 
}
```
or returning a pointer to distinguish between the error and contract not being deployed

Reply:
Also, do we have any tests to cover these cases?

Reply:
There can be two scenarios -
1. If Cap reg is present - is covered by in-memory test. We cannot restart memory nodes that's why had to deploy capreg first in a separate call so that we can start the memory nodes with capreg in node config.
2. If Cap reg is not present and deployed through DeployHomeChain - will be covered by docker tests( I will create a ticket to cover this in next pr) - In this case we will be restarting docker nodes once home chain is deployed with newly deployed cap reg address.